tests/tests/device.rs

```diff
@@ -91,3 +91,181 @@ async fn request_device_error_message() {
     }
     assert!(device_error.contains(expected), "{device_error}");
 }
+
+#[test]
+fn device_lose_then_more() {
+    // This is a test of device behavior after "lose the device". Specifically, all operations
+    // should trigger errors. To test this, it calls an explicit function to lose the device,
+    // which is not part of normal use and ideally wouldn't be exposed at all.
+    //
+    // TODO: Figure out how to make device.lose a private function only available to the test
+    // harness.
+    //
+    // On DX12 this test fails with a validation error in the very artifical actions taken
+    // after lose the device. The error is "ID3D12CommandAllocator::Reset: The command
```

Comment:
I think this is https://github.com/gfx-rs/wgpu/issues/3193, you can reference that here :)